crates/wdl-engine/src/backend/slurm_apptainer.rs

```diff
@@ -0,0 +1,879 @@
+        // node. If there is also an integer `hints: { gpu: n }`, we request `n`
+        // GPUs per node.
+        if let Some(true) = self
+            .spawn_request
+            .requirements()
+            .get(wdl_ast::v1::TASK_REQUIREMENT_GPU)
+            .and_then(Value::as_boolean)
+        {
+            match self.spawn_request.hints().get(wdl_ast::v1::TASK_HINT_GPU) {
+                Some(Value::Primitive(PrimitiveValue::Integer(n))) => {
+                    sbatch_command.arg(format!("--gpus-per-task={n}"));
+                }
+                Some(Value::Primitive(PrimitiveValue::String(hint))) => {
+                    warn!(
+                        %hint,
+                        "string hints for GPU are not supported; falling back to 1 GPU per host"
+                    );
+                    sbatch_command.arg("--gpus-per-task=1");
+                }
+                // Other hint value types should be rejected already, so the remaining valid case is
+                // a GPU requirement with no hints
+                _ => {
+                    sbatch_command.arg("--gpus-per-task=1");
+                }
+            }
+        }
+
+        // Add any user-configured extra arguments.
+        if let Some(args) = &self.backend_config.extra_sbatch_args {
+            sbatch_command.args(args);
+        }
+
+        sbatch_command
+            // Use verbose output that we can check later on
+            .arg("-v")
+            // Keep `sbatch` running until the job terminates
+            .arg("--wait")
+            // Pipe stdout and stderr so we can identify when a job begins, and can trace any other
+            // output. This should just be the `sbatch` verbose output on stderr.
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            // Name the Slurm job after the task ID, which has already been shortened to fit into
+            // the Slurm requirements.
+            .arg("--job-name")
+            .arg(&self.name)
+            // Send Slurm job stdout and stderr streams to these files. Since we redirect the
+            // Apptainer invocation's stdio to separate files, this will typically not contain
+            // anything, but can be useful for debugging if the scripts get modified.
+            .arg("-o")
+            .arg(slurm_stdout_path)
+            .arg("-e")
+            .arg(slurm_stderr_path)
+            // An explicit task count is required for some options
+            .arg("--ntasks=1")
+            // CPU request is rounded up to the nearest whole CPU
+            .arg(format!(
+                "--cpus-per-task={}",
+                self.required_cpu.ceil() as u64
+            ))
+            // Memory request is specified per node in megabytes; we round the request up to the
+            // next megabyte
+            .arg(format!(
+                "--mem={}",
+                (self.required_memory.as_u64() as f64 / bytesize::MIB as f64).ceil() as u64
```

Comment:
```suggestion
                (self.required_memory.as_u64() as f64 / bytesize::MB as f64).ceil() as u64
```

From the docs, the default units are in _megabytes_ (https://slurm.schedmd.com/sbatch.html).